extensions/efiler_api_client_auth.rb

```diff
@@ -0,0 +1,56 @@
+require 'sinatra/base'
+
+# See https://sinatrarb.com/extensions.html for more context about how this code is structured
+
+module Sinatra
+  module EfilerApiClientAuth
+    def authenticate_jwt
+      before do
+        verify_client_name_and_signature
+      end
+    end
+
+    def verify_client_name_and_signature
+      authorization_header = request.env['HTTP_AUTHORIZATION']
+      token = JWT::EncodedToken.new(authorization_header.delete_prefix("Bearer "))
+
+      allowed_client_names = Dir.children("client_public_keys").map do |public_key_filename|
+        File.basename(public_key_filename, ".pub")
+      end
+      token.verify_claims!(iss: allowed_client_names)
```

Comment:
why is this `allowed_client_names` instead of `allowed_client_name`? Could this be plural?